Src/Newtonsoft.Json/Utilities/ReflectionUtils.cs

```diff
@@ -123,13 +123,7 @@ public static MethodInfo GetBaseDefinition(this PropertyInfo propertyInfo)
                 return m.GetBaseDefinition();
             }
 
-            m = propertyInfo.GetSetMethod(true);
```

Comment:
The main part of the `GetBaseDefinition` static method could probably be reduced to:

```c#
return propertyInfo.GetGetMethod(true)?.GetBaseDefinition()
    ?? propertyInfo.GetSetMethod(true)?.GetBaseDefinition()
    ?? null;
```

And in the same vein, for the `IsVirtual` static method (just above it):

```c#
return propertyInfo.GetGetMethod(true)?.IsVirtual
    ?? propertyInfo.GetSetMethod(true)?.IsVirtual
    ?? false;
```